sacro/views.py

```diff
@@ -1,19 +1,76 @@
 import json
+from dataclasses import dataclass
+from functools import cached_property
 from pathlib import Path
+from urllib.parse import urlencode
 
 from django.conf import settings
-from django.http import Http404
+from django.http import FileResponse, Http404
 from django.template.response import TemplateResponse
+from django.urls import reverse
 
 
-def index(request):
-    path = request.GET.get("path")
-    if path is None:
+def reverse_qs(qs_dict, *args, **kwargs):
+    """Wrapper for django reverse that adds query parameters"""
+    url = reverse(*args, **kwargs)
+    return url + "?" + urlencode(qs_dict)
```

Comment:
Not sure how relevant this is so take this as more something to be aware of… `qs` in Django-land is typically a shortening of QuerySet, which is why we can't have nice things.